File: src/main/java/io/dddrive/doc/model/DocMeta.java

package io.dddrive.doc.model;

import java.util.List;

import io.dddrive.ddd.model.AggregateMeta;
import io.dddrive.doc.model.enums.CodeCaseDef;
import io.dddrive.doc.model.enums.CodeCaseStage;

public interface DocMeta extends AggregateMeta {

	List<DocPartTransition> getTransitionList();

	CodeCaseDef getCaseDef();

	CodeCaseStage getCaseStage();

	boolean isInWork();

	List<CodeCaseStage> getCaseStages();

	// availableActions: string[];

}

File: fm-dddrive/src/main/java/dddrive/app/doc/model/DocMeta.kt
package dddrive.app.doc.model

import dddrive.app.ddd.model.AggregateMeta
import dddrive.app.doc.model.enums.CodeCaseDef
import dddrive.app.doc.model.enums.CodeCaseStage
import java.time.OffsetDateTime

interface DocMeta : AggregateMeta {

	val docTypeId: String

	val caseDef: CodeCaseDef?
	val caseStages: List<CodeCaseStage>
	val caseStage: CodeCaseStage?
	val isInWork: Boolean
	// availableActions: string[];

	fun setCaseStage(
		caseStage: CodeCaseStage,
		userId: Any,
		timestamp: OffsetDateTime,
	)

	val transitionList: List<DocPartTransition>

}
